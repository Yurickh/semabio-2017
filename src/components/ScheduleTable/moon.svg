<?xml version="1.0" encoding="UTF-8"?>
<svg width="42px" height="39px" viewBox="0 0 42 39" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 45.2 (43514) - http://www.bohemiancoding.com/sketch -->
    <title>Fill 1</title>
    <desc>Created with Sketch.</desc>
    <defs>
        <rect id="path-1" x="0" y="0" width="56" height="42"></rect>
    </defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="BetterDesk" transform="translate(-757.000000, -2190.000000)">
            <g id="Fill-1" transform="translate(750.000000, 2187.000000)">
                <mask id="mask-2" fill="white">
                    <use xlink:href="#path-1"></use>
                </mask>
                <g id="Mask"></g>
                <path d="M29.6874492,4.116701 C30.1473507,3.82073575 30.6175103,3.55028475 31.0945085,3.30534799 C26.4356546,2.80186688 21.5784788,3.82753955 17.3402039,6.56436763 C7.56259445,12.8799939 4.78266911,25.8854553 11.1306782,35.6131867 C17.4769776,45.342619 30.5508331,48.1083632 40.3284426,41.792737 C44.5667174,39.0559089 47.4834164,35.0586771 48.9195402,30.6191984 C48.4989611,30.9542855 48.0612853,31.2723631 47.5996741,31.5700293 C39.9813793,36.4908767 29.7934488,34.3357735 24.8473701,26.7546406 C19.8995817,19.1735078 22.0674447,9.03924945 29.6874492,4.116701 Z" fill="#333F5D" mask="url(#mask-2)"></path>
            </g>
        </g>
    </g>
</svg>