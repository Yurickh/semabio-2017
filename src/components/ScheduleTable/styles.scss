@import 'ambience/color';

.schedule-table {
  display: flex;
  justify-content: center;

  height: 550px;
  margin-top: 33px;

  & > .-odd {
    align-self: flex-start;

    text-align: right;
  }

  & > .-even {
    align-self: flex-end;

    text-align: left;
  }

  & > .events > .event {
    line-height: 16px;

    margin-right: 35px;

    letter-spacing: -.25px;

    color: $color-tuna;
  }

  & > .events > .event > .time {
    font-size: 14px;

    margin-top: 60px;
  }

  & > .events > .event > .name {
    font-size: 18px;

    margin-top: 1em;
  }

  & > .times > .sun {
    position: relative;

    width: 150px;
    height: 3em;
    margin-top: 60px;
    z-index: 2;
  }

  & > .times > .sun::before {
    position: absolute;
    top: 0;
    left: 50%;
    transform: translateX(-50%);

    display: block;

    width: 45px;
    height: 45px;

    content: '';

    border-radius: 50%;
    background-color: $color-sun;
    z-index: 1;
  }

  & > .times > .sun.-small::before {
    margin-top: -18px;
  }

  & > .times > .sun.-medium::before {
    margin-top: -25px;
  }
}
