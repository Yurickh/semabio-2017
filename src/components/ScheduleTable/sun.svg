<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 21.0.2, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="sun" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 46 46" width="100%" height="100%" style="enable-background:new 0 0 46 46;" xml:space="preserve">
<style type="text/css">
	.st0{fill:#FFC612;}
</style>
<path class="part-4 st0" d="M4,12H42C38.2,5.4,31.1,1,23,1S7.8,5.4,4,12z"/>
<path class="part-3 st0" d="M1,23h44c0-4-1.1-7.8-3-11H4C2.1,15.2,1,19,1,23z"/>
<path class="part-2 st0" d="M4,34H42c1.9-3.2,3-7,3-11H1C1,27,2.1,30.8,4,34z"/>
<path class="part-1 st0" d="M42,34H4c3.8,6.6,10.9,11,19,11S38.2,40.6,42,34z"/>
</svg>
