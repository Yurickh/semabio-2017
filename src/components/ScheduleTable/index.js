import React, { Component } from 'react'
import PropTypes from 'prop-types'

import ReactSVG from 'react-svg'

import scheduleData from '../../helpers/scheduleData'

import sun from './sun.svg'

import './styles.css'

class ScheduleTable extends Component {
	scheduleData = scheduleData()
	oddTimes = ['small', 'full', 'medium', 'minimal']
	evenTimes = ['half', 'medium', 'small', 'moon']

	static propTypes = {
		selected: PropTypes.oneOf(Object.keys(scheduleData()))
	}

	schedule = () => {
		return this.scheduleData[this.props.selected]
	}

	eventsOfType = (type) => {
		// grab all the even or the odd indexes, depending on type
		const isOfType = index => index % 2 === +(type === 'even')
		return this.schedule().filter((_, index) => isOfType(index))
	}

	renderEvents = (type) => {
		const events = this.eventsOfType(type)

		return events.map(event => (
			<div className="event" key={`${event.time} - ${event.name}`}>
				<p className="time">{event.time}</p>
				<p className="name">{event.name}</p>
			</div>
		))
	}

	renderTimes = (type) => {
		let times = type === 'even' ? this.evenTimes : this.oddTimes

		return times
		.filter((time, index) => {
			// check if there's an event with given index present
			return Object.keys(this.eventsOfType(type)[index]).length
		})
		.map(time => (
			<div
				key={time}
				className={`sun -${time}`}
			>
				<ReactSVG path={sun} />
			</div>
		))
	}

	render() {
		return (
			<div className="schedule-table">
				<div className="events -odd">
					{this.renderEvents('odd')}
				</div>
				<div className="times -odd">
					{this.renderTimes('odd')}
				</div>
				<div className="times -even">
					{this.renderTimes('even')}
				</div>
				<div className="events -even">
					{this.renderEvents('even')}
				</div>
			</div>
		)
	}
}

export default ScheduleTable
