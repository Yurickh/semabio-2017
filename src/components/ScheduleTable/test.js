import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import ScheduleTable from '../ScheduleTable'

describe('ScheduleTable', () => {
  it('should render correctly', () => {
    const tree = shallow(<ScheduleTable />)
    expect(toJSON(tree)).toMatchSnapshot()
  })
})
