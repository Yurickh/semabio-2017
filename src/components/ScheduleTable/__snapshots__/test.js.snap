// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleTable should render correctly 1`] = `
<div
  className="schedule-table"
>
  <div
    className="events -odd"
  >
    <div
      className="event"
    >
      <p
        className="time"
      >
        08:00
      </p>
      <p
        className="name"
      >
        Minicursos Bloco A
      </p>
    </div>
    <div
      className="event"
    >
      <p
        className="time"
      >
        13:00
      </p>
      <p
        className="name"
      >
        A Odisseia da Vida
      </p>
    </div>
    <div
      className="event"
    >
      <p
        className="time"
      >
        15:45
      </p>
      <p
        className="name"
      >
        Coffee Break
      </p>
    </div>
    <div
      className="event"
    >
      <p
        className="time"
      >
        17:30
      </p>
      <p
        className="name"
      >
        Jantar
      </p>
    </div>
  </div>
  <div
    className="times -odd"
  >
    <div
      className="sun -small"
    >
      <ReactSVG
        callback={[Function]}
        className=""
        evalScripts="once"
        path="sun.svg"
        style={Object {}}
      />
    </div>
    <div
      className="sun -full"
    >
      <ReactSVG
        callback={[Function]}
        className=""
        evalScripts="once"
        path="sun.svg"
        style={Object {}}
      />
    </div>
    <div
      className="sun -medium"
    >
      <ReactSVG
        callback={[Function]}
        className=""
        evalScripts="once"
        path="sun.svg"
        style={Object {}}
      />
    </div>
    <div
      className="sun -minimal"
    >
      <ReactSVG
        callback={[Function]}
        className=""
        evalScripts="once"
        path="sun.svg"
        style={Object {}}
      />
    </div>
  </div>
  <div
    className="times -even"
  >
    <div
      className="sun -half"
    >
      <ReactSVG
        callback={[Function]}
        className=""
        evalScripts="once"
        path="sun.svg"
        style={Object {}}
      />
    </div>
    <div
      className="sun -medium"
    >
      <ReactSVG
        callback={[Function]}
        className=""
        evalScripts="once"
        path="sun.svg"
        style={Object {}}
      />
    </div>
    <div
      className="sun -small"
    >
      <ReactSVG
        callback={[Function]}
        className=""
        evalScripts="once"
        path="sun.svg"
        style={Object {}}
      />
    </div>
    <div
      className="sun -moon"
    >
      <ReactSVG
        callback={[Function]}
        className=""
        evalScripts="once"
        path="sun.svg"
        style={Object {}}
      />
    </div>
  </div>
  <div
    className="events -even"
  >
    <div
      className="event"
    >
      <p
        className="time"
      >
        11:30
      </p>
      <p
        className="name"
      >
        Almoço
      </p>
    </div>
    <div
      className="event"
    >
      <p
        className="time"
      >
        14:30
      </p>
      <p
        className="name"
      >
        Impactos do Ecoturismo
      </p>
    </div>
    <div
      className="event"
    >
      <p
        className="time"
      >
        16:15
      </p>
      <p
        className="name"
      >
        Ciência em Quadrinhos
      </p>
    </div>
    <div
      className="event"
    >
      <p
        className="time"
      >
        19:00 às 22:00
      </p>
      <p
        className="name"
      >
        Minicursos Bloco B
      </p>
    </div>
  </div>
</div>
`;
